From 6f28be8cb77df2b29a064612e26c4e5019b2b848 Mon Sep 17 00:00:00 2001
From: David Bauer <mail@david-bauer.net>
Date: Thu, 16 Mar 2023 20:02:00 +0530
Subject: [PATCH 159/500] OpenWrt:
 761-dt-bindings-net-add-RTL8152-binding-documentation.patch

dt-bindings: net: add RTL8152 binding documentation

Add binding documentation for the Realtek RTL8152 / RTL8153 USB ethernet
adapters.

Signed-off-by: David Bauer <mail@david-bauer.net>

Change-Id: I2daab7dbe08d15c73ca1a84f1231b0aad0f627dd
Signed-off-by: Ram Chandra Jangir <quic_rjangir@quicinc.com>
(cherry picked from commit 33522e7e092d8c844c3e41a61456e22152e69a88)
---
 .../bindings/net/realtek,rtl8152.yaml         | 36 +++++++++++++++++++
 1 file changed, 36 insertions(+)
 create mode 100644 Documentation/devicetree/bindings/net/realtek,rtl8152.yaml

diff --git a/Documentation/devicetree/bindings/net/realtek,rtl8152.yaml b/Documentation/devicetree/bindings/net/realtek,rtl8152.yaml
new file mode 100644
index 000000000000..7501b074a499
--- /dev/null
+++ b/Documentation/devicetree/bindings/net/realtek,rtl8152.yaml
@@ -0,0 +1,36 @@
+# SPDX-License-Identifier: GPL-2.0
+%YAML 1.2
+---
+$id: http://devicetree.org/schemas/net/realtek,rtl8152.yaml#
+$schema: http://devicetree.org/meta-schemas/core.yaml#
+
+title: Realtek RTL8152/RTL8153 series USB ethernet
+
+maintainers:
+  - David Bauer <mail@david-bauer.net>
+
+properties:
+  compatible:
+    oneOf:
+      - items:
+          - enum:
+              - realtek,rtl8152
+              - realtek,rtl8153
+
+  reg:
+    description: The device number on the USB bus
+
+  realtek,led-data:
+    description: Value to be written to the LED configuration register.
+
+required:
+  - compatible
+  - reg
+
+examples:
+  - |
+    usb-eth@2 {
+      compatible = "realtek,rtl8153";
+      reg = <2>;
+      realtek,led-data = <0x87>;
+    };
\ No newline at end of file
-- 
2.34.1

